storagemarket/impl/clientstates/client_states.go

```diff
@@ -136,35 +144,44 @@ func WaitingForDataRequest(ctx fsm.Context, environment ClientDealEnvironment, d
 	return ctx.Trigger(storagemarket.ClientEventDataTransferInitiated)
 }
 
-// VerifyDealResponse reads and verifies the response from the provider to the proposed deal
-func VerifyDealResponse(ctx fsm.Context, environment ClientDealEnvironment, deal storagemarket.ClientDeal) error {
-	resp, err := environment.ReadDealResponse(deal.ProposalCid)
+// CheckForDealAcceptance is run until the deal is sealed and published by the provider, or errors
+func CheckForDealAcceptance(ctx fsm.Context, environment ClientDealEnvironment, deal storagemarket.ClientDeal) error {
+	dealState, err := environment.GetProviderDealState(ctx.Context(), deal)
 	if err != nil {
-		return ctx.Trigger(storagemarket.ClientEventReadResponseFailed, err)
+		log.Warnf("error when querying provider deal state: %w", err) // TODO: at what point do we fail the deal?
```

Comment:
I'm thinking the proper thing here would be to have a maximum fail count.

Reply:
here's a suggestion: 
make waitAgain take an error value -> pass on to ClientEventWaitForDealState
in the FSM code, for if ClientEventWaitForDealState, if err != nil, also increment another value -- say "PollErrorCount".

in this code, add a check for deal.PollErrorCount > some constant (maybe 10? -- miner offline for 5 minutes seems bad) and if so, instead of waiting again, trigger ClientEventReadResponseFailed like before and add back in CheckForAcceptance as an acceptable transition state.

Reply:
I wonder about this; the Miner can be totally unresponsive to queries, but can still fulfill their side of the storage deal.  In that case, if we fail the deal on the Client but the Miner finishes sealing and publishing...